integration/clean/clean_dedicated_host/clean_dedicated_host.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"log"
+	"time"
+)
+
+// Can't release a host if it was being used within the last 24 hr add 2 hr as a buffer
+const keepDurationDedicatedHost = -1 * time.Hour * 26
+const tagName = "tag:Name"
+const tagValue = "IntegrationTestMacDedicatedHost"
+
+func main() {
+	err := cleanDedicatedHost()
+	if err != nil {
+		log.Fatalf("errors cleaning %v", err)
+	}
+}
+
+func cleanDedicatedHost() error {
+	log.Print("Begin to clean EC2 Dedicated Host")
+
+	expirationDateDedicatedHost := time.Now().UTC().Add(keepDurationDedicatedHost)
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		return err
+	}
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	dedicatedHosts, err := getDedicatedHost(cxt, ec2client)
+	if err != nil {
+		return err
+	}
+
+	dedicatedHostIds := make([]string, 0)
+	for _, dedicatedHost := range dedicatedHosts {
+		log.Printf("dedicated host id %v experation date %v dedicated host creation date raw %v",
+			*dedicatedHost.HostId, expirationDateDedicatedHost, *dedicatedHost.AllocationTime)
+		if expirationDateDedicatedHost.After(*dedicatedHost.AllocationTime) && dedicatedHost.State == types.AllocationStateAvailable {
+			log.Printf("Try to delete dedicated host %s tags %v launch-date %s", *dedicatedHost.HostId, dedicatedHost.Tags, *dedicatedHost.AllocationTime)
+			dedicatedHostIds = append(dedicatedHostIds, *dedicatedHost.HostId)
+		}
+	}
+
+	if len(dedicatedHostIds) == 0 {
+		log.Printf("No dedicated hosts to release")
+		return nil
+	}
+
+	releaseDedicatedHost := ec2.ReleaseHostsInput{HostIds: dedicatedHostIds}
+	_, err = ec2client.ReleaseHosts(cxt, &releaseDedicatedHost)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Didn't notice this the first time. We could just simplify this to `return err` because if there is an error, we would return a non-nil error, but if there is no error, returning `err` would just result in returning nil, which does the same thing as this code block.